### PR TITLE
[connectedmachine] release preview version 2023-10-03

### PR DESCRIPTION
**Pull request by yaotongms**
This release is targeting version 2023-10-03 of connectedmachine and using aaz. You will see all commands from the previous version with the following new commands:

- az connectedmachine run-command: Manage run commands on an Azure Arc-Enabled Server

-----
This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az.connectedmachine

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
connectedmachine

**Comment by yaotongms** [comment]
aaz PR: https://github.com/Azure/aaz/pull/349

**Comment by yaotongms** [comment]
There are no breaking changes in this release. The reported ones are just changes on flag names and removal of the flag 'location'.

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_create.py [hunk 1]
```diff
@@ -91,12 +91,12 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
         _args_schema = cls._args_schema
         _args_schema.auto_upgrade_minor_version = AAZBoolArg(
-            options=["--upgrade-minor-ver"],
+            options=["--auto-upgrade-minor", "--auto-upgrade-minor-version"],
             arg_group="Properties",
             help="Indicates whether the extension should use a newer minor version if one is available at deployment time. Once deployed, however, the extension will not upgrade minor versions unless redeployed, even with this property set to true.",
         )
         _args_schema.enable_automatic_upgrade = AAZBoolArg(
-            options=["--enable-auto-upgrade"],
+            options=["--auto-upgrade", "--enable-automatic-upgrade"],
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_create.py:99]: Let me check with you if this param is renamed and the current name `--enable-auto-upgrade` is deprecated? If so, it is better to add the current name back to aviod breaking change.
```suggestion
            options=["--enable-auto-upgrade", "--auto-upgrade", "--enable-automatic-upgrade"],
```

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_update.py [hunk 2]
```diff
@@ -87,13 +87,13 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
         _args_schema.auto_upgrade_minor_version = AAZBoolArg(
-            options=["--upgrade-minor-ver"],
+            options=["--auto-upgrade-minor", "--auto-upgrade-minor-version"],
             arg_group="Properties",
             help="Indicates whether the extension should use a newer minor version if one is available at deployment time. Once deployed, however, the extension will not upgrade minor versions unless redeployed, even with this property set to true.",
             nullable=True,
         )
         _args_schema.enable_automatic_upgrade = AAZBoolArg(
-            options=["--enable-auto-upgrade"],
+            options=["--auto-upgrade", "--enable-automatic-upgrade"],
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_update.py:96]: The same as above

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_update.py [hunk 3]
```diff
@@ -87,13 +87,13 @@ def _build_arguments_schema(cls, *args, **kwargs):
 
         _args_schema = cls._args_schema
         _args_schema.auto_upgrade_minor_version = AAZBoolArg(
-            options=["--upgrade-minor-ver"],
+            options=["--auto-upgrade-minor", "--auto-upgrade-minor-version"],
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_update.py:90]: The same as above

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_create.py (hunk 1 above)
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_create.py]: updated

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_update.py (hunk 2 above)
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_update.py]: updated

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/_upgrade_extension.py [hunk 4]
```diff
@@ -13,18 +13,19 @@
 
 @register_command(
     "connectedmachine upgrade-extension",
+    is_preview=True,
```
Comment (zhoxing-ms) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/_upgrade_extension.py:16]: May I ask why do you mark the GA command as `is_preview`? Please note that even if using the preview api-version for this command, if its interaction UX is stable for CLI customers, there is no need to mark it as `is_preview`, otherwise it may be a regression of stability for customers.

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/private_link_scope/_list.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
     "connectedmachine private-link-scope list",
 )
 class List(AAZCommand):
-    """Get a list of Azure Arc PrivateLinkScopes within a resource group. And Gets a list of all Azure Arc PrivateLinkScopes within a subscription.
+    """Get a list of Azure Arc PrivateLinkScopes within a resource                                          group. And Gets a list of all Azure Arc PrivateLinkScopes within a subscription.
```
Comment (zhoxing-ms) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/private_link_scope/_list.py:18]: Why did you add these spaces?

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/private_link_scope/_update.py [hunk 6]
```diff
@@ -15,10 +15,10 @@
     "connectedmachine private-link-scope update",
 )
 class Update(AAZCommand):
-    """Update an Azure Arc PrivateLinkScope. Note: You cannot specify a different value for InstrumentationKey nor AppId in the Put operation.
+    """Update an Azure Arc PrivateLinkScope. Note: You cannot                                          specify a different value for InstrumentationKey nor AppId in the Put operation.
```
Comment (zhoxing-ms) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/private_link_scope/_update.py:18]: Why did you add these spaces?

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py [hunk 7]
```diff
@@ -45,13 +45,13 @@ def test_machine_and_extension(self):
...
         self.cmd('az connectedmachine list -g {rg}', checks=[
-            self.check('length(@)', 1)
+            self.check('length(@)', 2)
         ])
 
         self.cmd('az connectedmachine extension create '
                 '--name "{customScriptName}" '
                 '--location "{location}" '
-                '--enable-auto-upgrade true '
+                '--enable-automatic-upgrade true '
```
Comment (zhoxing-ms) [inline src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py:54]: Why did you modify the name of the parameter here?

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py [hunk 8]
```diff
@@ -123,7 +123,7 @@ def test_machine_and_extension(self):
 
         self.cmd('az connectedmachine extension update '
                 '--name "{customScriptName}" '
-                '--enable-auto-upgrade false '
+                '--enable-automatic-upgrade false '
```
Comment (zhoxing-ms) [inline src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py:126]: Same as above

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_RunCommand_scenario.py [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+from azure.cli.testsdk.scenario_tests import (
+    live_only
+)
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+
+class RunCommandScenarioTest(ScenarioTest):
+    @live_only()
+    @ResourceGroupPreparer(name_prefix='cli_test_runcommand')
```
[the remark this answers is not in the document]
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/tests/latest/test_RunCommand_scenario.py:40]: Because some commands need to have subscription as input and are only valid for certain subscriptions. If I don't mark it as @live_only(), it fails on CI.

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/private_link_scope/_update.py (hunk 6 above)
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/private_link_scope/_update.py]: Not sure what happens when using aaz, but fixed.

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/_upgrade_extension.py (hunk 4 above)
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/_upgrade_extension.py]: I agree. Fixed to be stable instead.

**Review by zhoxing-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Comment by jonathan-ford** [comment]
There appears to be a bug in 'Remove-AzConnectedMachineRunCommand' aka 'Machine Run Commands - Delete', where the RunCommand is stuck in the ProvisioningStatus 'Deleting'.

https://github.com/MicrosoftDocs/azure-docs/issues/120126
